src/mcfret.c

```diff
@@ -0,0 +1,366 @@
+	              printf("Cluster trajectory file to short, could not fill buffer!!!\n");
+	              printf("ITIME %d\n",ti);
+	              exit(1);
+            }
+            /*Configuration belong to cluster*/ 
+        if (non->cluster==cl){
+	          Ncl++;
+        }
+        }
+        if (non->cluster==-1 || non->cluster==cl){
+      /*Initialize time-evolution operator*/ 
+            unitmat(vecr,non->singles);
+      /*Do projection on selected sites if asked*/
+            if (non->Npsites>0){
+                for (i=0;i<non->singles;i++){
+                    j=i*non->singles;
+                    projection(vecr+j,non);
+                }
+            }
+            clearvec(veci,non->singles*non->singles);
+
+      /*Loop over delay*/ 
+            for (t1=0;t1<non->tmax;t1++){
+	              tj=ti+t1;
+	    /* Read Hamiltonian */
+	              if (!strcmp(non->hamiltonian,"Coupling")){
+                    if (read_Dia(non,Hamil_i_e,H_traj,tj)!=1){
+                        printf("Hamiltonian trajectory file to short, could not fill buffer!!!\n");
+                        exit(1);
+                    }
+                }
+                else {
+	                  if (read_He(non,Hamil_i_e,H_traj,tj)!=1){
+	                      printf("Hamiltonian trajectory file to short, could not fill buffer!!!\n");
+	                      exit(1);
+  	                }
+                }
+             /* Call the MCFRET Response for Absorption matrix */
+                if (!strcmp(non->technique, "MCFRET") || !strcmp(non->technique, "MCFRET-Absorption")  ){
+                    printf("Calculation Absortion Matrix for MCFRET.\n");
+                    mcfret_response_function(Hamil_i_e, re_S_Abs, im_S_Abs,tj,non,vecr,veci, dens_mat,0);
+        
+                }
+
+            /* Call the MCFRET Response for Emission matrix */
+                if (!strcmp(non->technique, "MCFRET") || !strcmp(non->technique, "MCFRET-Emission")  ){
+                    printf("Calculation Emssion Matrix for MCFRET.\n");
+                    mcfret_response_function(Hamil_i_e, re_S_Emi, im_S_Emi,tj,non,vecr,veci, dens_mat,1);
+                }
+                #pragma omp parallel for
+	              for (i=0; i<non->singles; i++){
+	                  j=i*non->singles;
+	              /* Propagate vector*/
+	                  if (non->propagation==1) propagate_vec_coupling_S(non,Hamil_i_e,vecr+j,veci+j,non->ts,1);
+	                      if (non->propagation==0){
+	                          if (non->thres==0 || non->thres>1){
+	                              propagate_vec_DIA(non,Hamil_i_e,vecr+j,veci+j,1);
+	                          } 
+                        else {
+	                          elements=propagate_vec_DIA_S(non,Hamil_i_e,vecr+j,veci+j,1);
+	                          if (samples==non->begin){
+	                              if (t1==0){
+		                                if (x==0){
```

Comment:
It seems odd to include print statements inside the omp parallel part. Is it not better to do this before the omp loop?